File: VB/GrowthWare.Framework.BusinessData/DataAccessLayer/MySQL/V5_6_21/DClientChoices.vb
Imports GrowthWare.Framework.BusinessData.DataAccessLayer.MySQL.Base
Imports GrowthWare.Framework.BusinessData.DataAccessLayer.Interfaces
Imports MySql.Data.MySqlClient

Namespace DataAccessLayer.MySQL.V5_6_21
    ''' <summary>
    ''' DClientChoices provides all database interaction to SQL Server 2000 to 2005
    ''' </summary>
    ''' <remarks>
    '''		The Profile and SecurityEntitySeqID properties must be set
    '''		before using any methods.
    '''		Properties where chosen instead of parameters because all
    '''		methods will need one or both to perform their work.
    '''	</remarks>
    Public Class DClientChoices
        Inherits DDBInteraction
        Implements IDClientChoices

#Region "Public Methods:"
        Public Function GetChoices(ByVal account As String) As DataRow Implements IDClientChoices.GetChoices
            If String.IsNullOrEmpty(account) Then Throw New ArgumentException("Must set the Account property", "account")
            Dim myParameters() As MySqlParameter =
            {
              New MySqlParameter("@P_ACCOUNT", account),
              New MySqlParameter("@P_Debug", 0)
            }
            Return MyBase.GetDataRow("ZGWCoreWeb.Get_Account_Choice", myParameters)
        End Function

        <System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Design", "CA1062:Validate arguments of public methods", MessageId:="0")>
        Public Sub Save(ByVal clientChoicesStateHashtable As Hashtable) Implements IDClientChoices.Save
            If clientChoicesStateHashtable Is Nothing Or clientChoicesStateHashtable.Count = 0 Then Throw New ArgumentNullException("clientChoicesStateHashtable", "Must set the clientChoicesStateHashTable property")
            Dim HashKeyEnum As IEnumerator = CType(clientChoicesStateHashtable.Keys, IEnumerable).GetEnumerator()
            Dim HashValEnum As IEnumerator = CType(clientChoicesStateHashtable.Values, IEnumerable).GetEnumerator()
            Dim commandParameters(clientChoicesStateHashtable.Count - 1) As MySqlParameter
            Dim x As Integer = 0
            Do While (HashKeyEnum.MoveNext() And HashValEnum.MoveNext())
                Dim myParameter As MySqlParameter = New MySqlParameter("@P_" + HashKeyEnum.Current.ToString(), SqlDbType.NVarChar, 1000)
                myParameter.Value = HashValEnum.Current.ToString()
                commandParameters.SetValue(myParameter, x)
                x = x + 1
            Loop
            MyBase.ExecuteNonQuery("ZGWCoreWeb.Set_Account_Choices", commandParameters)
        End Sub
#End Region
    End Class
End Namespace

File: Net_Core/DataAccess/SQLServer/DClientChoices.cs
using GrowthWare.DataAccess.Interfaces;
using GrowthWare.DataAccess.SQLServer.Base;
using System;
using System.Collections;
using System.Data;
using System.Data.SqlClient;

namespace GrowthWare.DataAccess.SQLServer;

/// <summary>
/// DClientChoices provides all database interaction to SQL Server 2008
/// </summary>
/// <remarks>
///		The Profile and SecurityEntitySeqID properties must be set
///		before using any methods.
///		Properties where chosen instead of parameters because all
///		methods will need one or both to perform their work.
///	</remarks>
public class DClientChoices : AbstractDBInteraction, IClientChoices
{

#region Constructors
    public DClientChoices(string connectionString): base()
    {
        this.ConnectionString = connectionString;
    }
#endregion

#region Pubic Methods
    DataRow IClientChoices.GetChoices(string account)
    {
        string mStoredProcedure = "ZGWCoreWeb.Get_Account_Choice";
        if (String.IsNullOrEmpty(account)) { throw new ArgumentException("Must set the Account property", nameof(account)); };
        SqlParameter[] myParameters =
        {
            new SqlParameter("@P_ACCOUNT", account)
        };
        return base.GetDataRow(mStoredProcedure, myParameters);
    }

    void IClientChoices.Save(Hashtable clientChoicesStateHashtable)
    {
        if (clientChoicesStateHashtable == null || clientChoicesStateHashtable.Count == 0) { throw new ArgumentNullException(nameof(clientChoicesStateHashtable), "Must set the clientChoicesStateHashTable property"); };
        string mStoredProcedure = "ZGWCoreWeb.Set_Account_Choices";
        IEnumerator HashKeyEnum = ((IEnumerable)clientChoicesStateHashtable.Keys).GetEnumerator();
        IEnumerator HashValEnum = ((IEnumerable)clientChoicesStateHashtable.Values).GetEnumerator();
        SqlParameter[] commandParameters = new SqlParameter[clientChoicesStateHashtable.Count];
        int x = 0;
        while ((HashKeyEnum.MoveNext() & HashValEnum.MoveNext()))
        {
            SqlParameter myParameter = new SqlParameter("@P_" + HashKeyEnum.Current.ToString(), SqlDbType.NVarChar, 1000);
            myParameter.Value = HashValEnum.Current.ToString();
            commandParameters.SetValue(myParameter, x);
            x = x + 1;
        }
        base.ExecuteNonQuery(mStoredProcedure, commandParameters);
    }
#endregion

}
